### ISSUE TITLE
Docstrings of IMAP4Client.status and IMAP4Client.fetch don't give enough information to be able to use those APIs

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3475]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3475|
|Type|defect|
|Created|2008-10-16 13:21:49Z|

The documentation for what each of these methods returns is quite fuzzy.  It should be more specific, describing actual types and, where dictionaries are involved, what to expect as the keys and values.

Attachments:

* [imap4-doc.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/260/260c4f07e206b9af71e32d87d5cb089d497289d3/4ff7b182435e2831572f68fc861865a3e4459bac.patch) (2858 bytes) - added by exarkun on 2008-10-16 13:22:32Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__3475 3475
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__documentation documentation
time__1224163309000000 1224163309000000
changetime__1224441702000000 1224441702000000
version__None None
owner__ 
cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_2"></a>|@ralphm set owner to @exarkun|
|-|-|

Looks good to me. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

(In [24988]) Apply imap4-doc.patch

Author: exarkun
Reviewer: ralphm
Fixes: [#3475](https://github.com/twisted/twisted/issues/3475)

Improve the docstrings for `IMAP4Client.status` and `IMAP4Client.fetch`.
The docstrings now describe the return value more completely.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
